data-ingestion-service/src/main/java/gov/cdc/dataingestion/kafka/integration/service/KafkaConsumerService.java

```diff
@@ -193,6 +196,48 @@ public void handleMessageForRawElr(String message,
         validationHandler(message, hl7ValidationActivated, dataProcessingEnable);
     }
 
+
+    /**
+     * Receive XML converted HL7 and save it to NBS Interface
+     * Description: The ELR ingestion endpoint that we have right now is modified to accept both
+     * plain text HL7 as well as HL7s converted to XML already.
+     * */
+    @RetryableTopic(
+            attempts = "${kafka.consumer.max-retry}",
+            autoCreateTopics = "false",
+            dltStrategy = DltStrategy.FAIL_ON_ERROR,
+            retryTopicSuffix = "${kafka.retry.suffix}",
+            dltTopicSuffix = "${kafka.dlt.suffix}",
+            // retry topic name, such as topic-retry-1, topic-retry-2, etc
+            topicSuffixingStrategy = TopicSuffixingStrategy.SUFFIX_WITH_INDEX_VALUE,
+            // time to wait before attempting to retry
+            backoff = @Backoff(delay = 1000, multiplier = 2.0),
+            // if these exceptions occur, skip retry then push message to DLQ
+            exclude = {
+                    SerializationException.class,
+                    DeserializationException.class,
+                    DuplicateHL7FileFoundException.class,
+                    DiHL7Exception.class,
+                    HL7Exception.class,
+                    XmlConversionException.class,
+                    JAXBException.class
+            }
+
+    )
+    @KafkaListener(
+            topics = "${kafka.raw.xml-topic}"
+    )
+    public void handleMessageForElrXml(String message,
+                                       @Header(KafkaHeaders.RECEIVED_KEY) String messageId,
+                                       @Header(KafkaHeaders.RECEIVED_TOPIC) String topic,
+                                       @Header(KafkaHeaderValue.DATA_PROCESSING_ENABLE) String dataProcessingEnable) {
+        log.debug(topicDebugLog, messageId, topic);
+
+        boolean dataProcessingApplied = Boolean.parseBoolean(dataProcessingEnable);
+        NbsInterfaceModel nbsInterfaceModel = nbsRepositoryServiceProvider.saveElrXmlMessage(messageId, message, dataProcessingApplied);
+        log.debug("Saved Elr xml to NBS_interface table with uid: {}", nbsInterfaceModel.getNbsInterfaceUid());
+    }
```

Comment:
Did you send out any message after saving to NBS? If you did send the message to the un_process topic, then I probably missed it somewhere. However, if you have not, can you add another producing logic here (it should only happen when version is 2)?

should be similar to the code at line 502
https://github.com/CDCgov/NEDSS-DataIngestion/blob/main/data-ingestion-service/src/main/java/gov/cdc/dataingestion/kafka/integration/service/KafkaConsumerService.java